platform/web/os_web.cpp

```diff
@@ -247,6 +247,58 @@ bool OS_Web::is_userfs_persistent() const {
 	return idb_available;
 }
 
+Error OS_Web::move_to_trash(const String &p_path) {
+	String path = p_path.rstrip("/"); // Strip trailing slash when path points to a directory.
+
+	if (!is_userfs_persistent()) {
+		// If the file system is not persistent, just remove the file.
+		DirAccess::remove_file_or_error(path);
+		return OK;
+	}
+
+	String trash_path = "/home/web_user/.trash";
+
+	// Create trash directory if it doesn't exist.
+	Ref<DirAccess> dir_access = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+	Error err = dir_access->make_dir_recursive(trash_path);
+	ERR_FAIL_COND_V_MSG(err != OK, err, "Could not create the trash path \"" + trash_path + "\"");
+
+	// The trash can is successfully created, now we check that we don't exceed our file name length limit.
+	// If the file name is too long trim it so we can add the identifying number.
+	// Assumes that the file name length limit is 255 characters.
+	String file_name = path.get_file();
+	if (file_name.length() > 240) {
+		file_name = file_name.substr(0, file_name.length() - 15);
+	}
```

Comment:
If the id number is 99 at most, shouldn't the length check be 253 characters instead?

Reply:
Ah, that's correct! I took this code directly without any modification from [platform/linuxbsd/os_linuxbsd.cpp](https://github.com/godotengine/godot/blob/master/platform/linuxbsd/os_linuxbsd.cpp#L1081-L1087). I'll make sure to update it accordingly. Thanks for pointing that out!